src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts

```diff
@@ -141,6 +164,30 @@ class TerminalSuggestContribution extends DisposableStore implements ITerminalCo
 		}
 	}
 
+	private async _loadLspCompletionAddon(xterm: RawXtermTerminal): Promise<void> {
+		// Load and register the LSP completion providers (one per language server)
+		this._lspModelProvider.value = this._instantiationService.createInstance(LspTerminalModelContentProvider);
```

Comment:
we should pass in the terminal capabilities here so we have access to command detection etc